### PR TITLE
Allow passing envvar into build process to build base image for a spe…

### PR DESCRIPTION
**Pull request by munkyshi**
…cific python version

Using this to provide CircleCI builds with the correct base sqitch image for varying python versions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,12 +7,13 @@ Kolena
 Steps for building Kolena sqitch images:
 
 ```sh
-./build
-docker tag sqitch/sqitch:1.1.0 ghcr.io/kolenaio/docker-sqitch/sqitch:1.1.0
-docker push ghcr.io/kolenaio/docker-sqitch/sqitch:1.1.0
-env DIR=snowflake ./build --build-arg sf_account=ara65239
-docker tag sqitch/sqitch:1.1.0-snowflake ghcr.io/kolenaio/docker-sqitch/sqitch:1.1.0-snowflake
-docker push ghcr.io/kolenaio/docker-sqitch/sqitch:1.1.0-snowflake
+for py_version in 3.7 3.8 3.9 3.10
```
Comment (gordonhart) [inline README.md]: Also 3.11?

**Review by gordonhart — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (gordonhart) [reply README.md]: Added and resolved merge conflicts with #6.

File: build [hunk 2]
```diff
@@ -48,6 +45,8 @@ docker buildx build --platform=linux/amd64 --pull \
   --label org.opencontainers.image.ref.name="sqitch${PKG}-${VERSION}" \
   --label org.opencontainers.image.title="Sqitch" \
   --label org.opencontainers.image.description="Sensible database change management" \
+  --platform linux/amd64 \
```
Comment (gordonhart) [inline build]: Hmm, I think this is [breaking CI](https://github.com/kolenaIO/docker-sqitch/actions/runs/4680510669/jobs/8291948630) — can we omit?

Comment (munkyshi) [reply build]: I think so -- looks like this is already specified above now.

File: build [hunk 3]
```diff
@@ -26,13 +28,8 @@ tagopt=(
     --tag "${REGISTRY}/${NAME}:${VERSION}${PKG}"
 )
 
-# Include the git tag if differnt from the version.
-if [ "$GITHUB_REF_TYPE" == "tag" ]; then
-    GIT_TAG=${GITHUB_REF/refs\/tags\//}
-    if [ "$GIT_TAG" != "$VERSION" ]; then
-        tagopt+=(--tag "${REGISTRY}/${NAME}:${GIT_TAG}${PKG}")
-    fi
-fi
```
Comment (gordonhart) [inline build:35]: Hard for me to see what's going on with this diff — do we want to keep this?

File: build [hunk 4]
```diff
@@ -6,7 +6,9 @@ NAME=sqitch
 VERSION=1.3.1
 
 DIR=${DIR:=.}
-REGISTRY=${REGISTRY:="$NAME"}
+PY_VERSION=${PY_VERSION:="3.9"}
+REGISTRY=${REGISTRY:="ghcr.io/kolenaio/docker-sqitch"}
+GIT_BRANCH=${GIT_BRANCH:-${TRAVIS_BRANCH:-$(git rev-parse --abbrev-ref HEAD)}}
```
Comment (gordonhart) [inline build:11]: Remove? Looks unused

Comment (munkyshi) [reply build:11]: Whoops, will do in follow up

File: build (hunk 3 above)
Comment (munkyshi) [reply build:35]: It's not useful to us, though it doesn't hurt to keep around to minimize the diff

**Pull request merged**